src/quicklist.c

```diff
@@ -1728,15 +1728,15 @@ int quicklistTest(int argc, char *argv[], int accurate) {
 
         TEST("add to tail of empty list") {
             quicklist *ql = quicklistNew(-2, options[_i]);
-            quicklistPushTail(ql, "hello", 6);
+            quicklistPushTail(ql, "hello", 5);
```

Comment:
i think maybe the author intended to add the null terminator as well...
any specific reason to remove it now?

Reply:
Is it possible. But i don't think there should counting the NULL... (I simply belive it was a typo error. maybe)
This is also used in other tests. Although I’m a bit curious, if I give a big number, what the result will be
```c
        TEST_DESC("rotate 500 val 5000 times at compress %d", options[_i]) {
            for (int f = 0; f < fill_count; f++) {
                quicklist *ql = quicklistNew(fills[f], options[_i]);
                quicklistPushHead(ql, "900", 3);
                quicklistPushHead(ql, "7000", 4);
                quicklistPushHead(ql, "-1200", 5);
                quicklistPushHead(ql, "42", 2);

        TEST_DESC("insert once in elements while iterating at compress %d",
                  options[_i]) {
            for (int f = 0; f < fill_count; f++) {
                quicklist *ql = quicklistNew(fills[f], options[_i]);
                quicklistPushTail(ql, "abc", 3);

        zl = ziplistPush(zl,(unsigned char*)"quux",4,ZIPLIST_TAIL);
```

Reply:
i don't understand why you wanted to indicate when you posted that big block of code.
please explain..

to me it seems obvious that the author intended to include the null terminator in these pushes, they're all using one extra byte than the length of the string.
if there's no real need to change these, i'd rather not change them.

Reply:
emm... The reason i posted this big block of code is that i want to show the other cases usage...
Sorry for the delay. I also try to find out the original intention of the author.. (but failed)
Yes. Including the NULL, the test work fine so indeed no need to change it. I reverted it.
I also do not have the strong reason to do that.

hate/sorry that such small fix has troubled you...